plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java

```diff
@@ -146,4 +162,47 @@ public void close()
     {
         dockerContainer.close();
     }
+
+    public static class DatabaseEventsRecorder
+    {
+        private final TestingPostgreSqlServer server;
+        private final int startingFromEventsCount;
+        private final int lastEventsCount;
+
+        private DatabaseEventsRecorder(TestingPostgreSqlServer server)
+        {
+            this(server, server.getRemoteDatabaseEvents().size(), Integer.MAX_VALUE);
+        }
+
+        private DatabaseEventsRecorder(TestingPostgreSqlServer server, int startingEventsCount, int lastEventCount)
+        {
+            this.server = requireNonNull(server, "server is null");
+            this.startingFromEventsCount = startingEventsCount;
+            this.lastEventsCount = lastEventCount;
+        }
+
+        public DatabaseEventsRecorder stopEventsRecording()
+        {
+            return new DatabaseEventsRecorder(this.server, startingFromEventsCount, this.server.getRemoteDatabaseEvents().size());
+        }
+
+        public Stream<String> streamQueriesContaining(String queryPart, String... alternativeQueryParts)
+        {
+            ImmutableList<String> queryParts = ImmutableList.<String>builder()
+                    .add(queryPart)
+                    .addAll(ImmutableList.copyOf(alternativeQueryParts))
+                    .build();
+            return streamQueries()
+                    .filter(query -> queryParts.stream().anyMatch(query::contains));
+        }
+
+        private Stream<String> streamQueries()
+        {
+            return server.getRemoteDatabaseEvents()
+                    .stream()
+                    .skip(startingFromEventsCount)
+                    .limit(lastEventsCount)
```

Comment:
Should this be the difference between the starting and lastEventCount ? 
BTW Do we need this bottom limit, since the test are single threaded and we already have a starting point.

Reply:
It was a request from @kokosing to make it immutable. Thanks to this it is, as every method will always return same events. We are not using this fact at this moment in our tests, but it's still better to have it as such and don't be forced to think about it in the tests.

Reply:
In this case, should limit be the difference between the startingEvent and lastEvents or should we re-order the streams ?

Reply:
So I believe that this is correct, as skip doesn't truncate the stream, it just skips first few and limit limits all values, but to be honest, I would prefer to have a test for that to be 100% sure, and we don't have that here. Due to that, I've decided to rewrite this part quickly, to not confuse anyone and to have immutability down to the values identity.